### PR TITLE
Allow missing columns for mongo storage

### LINKED ISSUE #26490: MongoDB engine can return null value when  some record lack of property in mongo collection?
How to reproduce：
1.docker image : clickhouse-server : 21.6.8.62
2.CREATE TABLE statements for all tables involved
CREATE TABLE IF NOT EXISTS mongo_table
(
_id UInt64,
updated_time DateTime,
account Nullable(String)
)
ENGINE = MongoDB(':', '<DB_NAME>', '<COLLECTION_NAME>', '', '')

3.Queries to run that lead to unexpected result

select * from mongo_table
Received exception from server (version 21.6.8):
Code: 10. DB::Exception: Received from localhost:9000. DB::Exception: Column account is absent in MongoDB collection: While executing SourceFromInputStream.

i found some record have no "account " in mongodb collection . i epxect null value when some record have no "account "

### LINKED ISSUE #36119: MongoDB engine question - how to query to schemaless
Hi, I am testing mongodb engine to get data from mongodb.

As you know, Mongodb is schemaless.
so, some documents don't have all fields. 
but, I want to get data from the document with blank or null.

below is an example.

mongo
```
db.chu.find()
{ "_id" : ObjectId("6253d4e4fe1ad41d94d9f04c"), "id" : 1, "c1" : "abc" }
{ "_id" : ObjectId("6253d4ecfe1ad41d94d9f04d"), "id" : 2, "c2" : "hello" }
{ "_id" : ObjectId("6253d502fe1ad41d94d9f04e"), "id" : 2, "c1" : "efg", "c2" : "hi" }
```

CH
```sql
CREATE TABLE mongo_table
(
    `id` UInt64,
    `c1` String,
    `c2` String
)
ENGINE = MongoDB('127.0.0.1:30002', 'test', 'chu', 'yhchu', '123456')

select * from mongo_table;

SELECT *
FROM mongo_table

↑ Progress: 0.00 rows, 0.00 B (0.00 rows/s., 0.00 B/s.) 
Received exception from server (version 20.6.4):
Code: 10. DB::Exception: Received from localhost:9000. DB::Exception: Column `c2` is absent in MongoDB collection: While executing SourceFromInputStream. 
```
how can I query that table?
Could you give me any clues?

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Allow missing columns for mongo storage. Closes https://github.com/ClickHouse/ClickHouse/issues/36119. Closes https://github.com/ClickHouse/ClickHouse/issues/26490.

### CODE REVIEW

File: src/Processors/Transforms/MongoDBSource.cpp [hunk 1]
```diff
@@ -341,17 +342,20 @@ Chunk MongoDBSource::generate()
             for (const auto idx : collections::range(0, size))
             {
                 const auto & name = description.sample_block.getByPosition(idx).name;
+                bool is_nullable = description.types[idx].second;
 
-                if (strict_check_names && !document->exists(name))
+                if (!is_nullable && strict_check_names && !document->exists(name))
```
Comment (CurtizJ) [inline src/Processors/Transforms/MongoDBSource.cpp:347]: Why not to allow inserting defaults for missing non-nullable columns?

Comment (kssenii) [reply src/Processors/Transforms/MongoDBSource.cpp:347]: makes sense

**Comment by kssenii** [comment]
@Mergifyio update

**Review by CurtizJ — APPROVED** [review]

**Comment by kssenii** [comment]
Integration tests:
```
E               Exception: Command ['docker-compose', '--env-file', '/ClickHouse/tests/integration/test_async_drain_connection/_instances_0/.env', '--project-name', 'roottestasyncdrainconnection', '--file', '/ClickHouse/tests/integration/test_async_drain_connection/_instances_0/node/docker-compose.yml', 'pull'] return non-zero code 1: Pulling node ... 
E               Pulling node ... error
E               
E               ERROR: for node  Get "https://registry-1.docker.io/v2/": context deadline exceeded
E               Get "https://registry-1.docker.io/v2/": context deadline exceeded
```

**Pull request merged**